catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/action/SourceActionProviderImpl.java

```diff
@@ -88,10 +92,14 @@ public void setDescription(String description) {
 
   private Action toAction() {
     try {
-      return new ActionImpl(getId(), title, description, new URL(url));
+      return new ActionImpl(getId(), title, description, new URL(constructUrl()));
     } catch (MalformedURLException e) {
       LOGGER.debug("Unable to parse the action url: url={}", url, e);
       return null;
     }
   }
+
+  protected String constructUrl() {
```

Comment:
Do we need this when we already have `getUrl()`?

Reply:
Added javadoc.